### PR TITLE
Introduce a null promotion configuration

### PR DESCRIPTION
**Pull request by mamhoff**
## Summary

This adds a "null" promotion configuration, consisting of only those extension points for the promotion system that are used from within core (rather than from within the promotion system).

This is a nice exercise as it also tells us what extension points we need for any promo system: A class to help recalculate orders, and one to add coupon codes to orders. 

This includes the commits from #5635 . I'll rebase once that is merged.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: core/app/models/spree/null_adjuster.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Spree
+  class NullAdjuster
```
Comment (tvdeyen) [inline core/app/models/spree/null_adjuster.rb:4]: Maybe a more specific name would help to understand the purpose of the object.

What about?

```rb
Spree::NullPromotionAdjuster
```

**Review by tvdeyen — COMMENTED** [review]
Very nice work!

**Comment by tvdeyen** [comment]
@mamhoff the PR title is a bit confusing, though. Mind to adjust it?

File: core/app/models/spree/null_adjuster.rb (hunk 1 above)
Comment (kennyadsl) [reply core/app/models/spree/null_adjuster.rb:4]: I second this question, unless you see a reason to keep it more generic. In that case, can you please help us understand where it is?

**Review by kennyadsl — COMMENTED** [review]
Well done, thanks Martin, left a comment.

File: core/app/models/spree/null_adjuster.rb (hunk 1 above)
Comment (mamhoff) [reply core/app/models/spree/null_adjuster.rb]: My observation was that the class doesn't actually do anything promotion-specific, and the interface (`#call`) also doesn't necessarily do anything. I was suspecting (but wrongly) that also tax adjuster might share that interface. 

Whatever: It's completely fine to call it `NullPromotionAdjuster`, and I've amended the PR to reflect that.

**Review by jarednorman — APPROVED** [review]
Great work. 👏

**Review by kennyadsl — APPROVED** [review]
👍 Thanks Martin, you rock!

**Pull request merged**